projects/addon-mobile/directives/ripple/ripple.directive.ts

```diff
@@ -41,60 +35,108 @@ class TuiRippleStyles {}
+        if (!tuiIsHTMLElement(element)) {
+            return;
+        }
 
-        this.start$
-            .pipe(
-                mergeMap(ripple => {
-                    const animationEndOn$ = tuiTypedFromEvent(ripple, 'animationend');
+        const ripple = this.createRipple(x, y, element.getBoundingClientRect());
+        const touchEnd$ = merge(
+            tuiTypedFromEvent(element, 'pointerup'),
+            tuiTypedFromEvent(element, 'pointercancel'),
+        );
+        const touchMove$ = tuiTypedFromEvent(element, 'pointermove');
+
+        element.appendChild(ripple);
+
+        const animationEndOn$ = tuiTypedFromEvent(
+            ripple.animate(
+                [
+                    {
+                        transform: 'scale(0)',
+                        opacity: '0.12',
+                    },
+                    {
+                        opacity: '0.12',
+                    },
+                ],
+                {
+                    duration: this.duration,
+                },
+            ),
+            'finish',
+        );
 
-                    return race(
-                        timer(TOUCH_MOVE_DELAY).pipe(map(TUI_FALSE_HANDLER)),
-                        touchEnd$.pipe(map(TUI_TRUE_HANDLER)),
-                    ).pipe(
-                        take(1),
-                        // eslint-disable-next-line rxjs/no-unsafe-takeuntil
-                        takeUntil(touchMove$),
-                        tap(() => {
-                            this.renderer.setStyle(
-                                ripple,
-                                'background',
-                                this.tuiRipple || null,
-                            );
-                            this.renderer.appendChild(this.el, ripple);
-                        }),
-                        switchMap(isTap =>
-                            isTap
-                                ? animationEndOn$
-                                : race(
-                                      touchEnd$.pipe(switchMap(() => animationEndOn$)),
-                                      animationEndOn$.pipe(switchMap(() => touchEnd$)),
-                                  ),
+        race(
+            touchEnd$.pipe(switchMap(() => animationEndOn$)),
+            animationEndOn$.pipe(switchMap(() => touchEnd$)),
+        )
+            .pipe(
+                first(),
+                takeUntil(touchMove$),
+                defaultIfEmpty(undefined),
+                switchMap(() =>
+                    tuiTypedFromEvent(
+                        ripple.animate(
+                            [
+                                {
+                                    opacity: '0.12',
+                                },
+                                {
+                                    opacity: '0',
+                                },
+                            ],
+                            {
+                                duration: this.duration,
```

Comment:
Looks like we can even save `private readonly options = { duration: tuiGetDuration(...) }` :)

Maybe also save `TO_KEYFRAMES` and `FROM_KEYFRAMES` to constants and then this whole stream even more readable